app/controllers/security_events_controller.rb

```diff
@@ -26,6 +32,21 @@ def show
     @security_event = security_event
   end
 
+  def search
+    email = params[:email]
+
+    if email.present?
+      if (user = User.find_by(email: email))
+        redirect_to security_events_all_path(user_uuid: user.uuid)
```

Comment:
My feeling is that emails don't belong in URLs, so that's why the this gets turned into a UUID

Reply:
Also that GET url params are easier to work with for a search, instead of doing POST body to accommodate an email

Reply:
Not major, but any reason we couldn't do this in the existing `all` route?

Reply:
you mean `POST /security_events/all => security_events#search`? Yeah that makes sense

Reply:
Sorry, I mean something like `GET /security_events/all?email=mitchell.henke@gsa.gov` redirects to `GET /security_events/all?user_uuid=abcd1234`. Should we need to support more field searching in the future, it becomes along the lines of:

```ruby
def all
  email = params.delete(:email)
  if email.present?
    user = User.find_by(uuid: params[:user_uuid])
    params[:user_uuid] = user&.uuid
    redirect_to security_events_all_path(params) and return
  end
# ...
end
```

The flash could still be added in there somewhere.

Reply:
If we did a GET like that, the email address from the URL is still in the logs, which is what I was trying to avoid

Reply:
ahhh, got it